src/Kliva/Models/ActivitySummaryService.cs

```diff
@@ -0,0 +1,83 @@
+using System;
+using System.Reactive.Disposables;
+using System.Reactive.Linq;
+using DynamicData;
+using DynamicData.Binding;
+using GalaSoft.MvvmLight.Threading;
+using Kliva.Services.Interfaces;
+
+namespace Kliva.Models
+{
+    public class ActivitySummaryService : IDisposable
+    {        
+        private readonly IStravaAthleteService _stravaAthleteService;
+        private readonly SourceCache<ActivitySummary, long> _friendsActivitySummaryCache;
+        private readonly SourceCache<ActivitySummary, long> _myActivitySummaryCache;
+        private readonly ActivitySummaryCacheLoader _friendsCacheLoader;
+        private readonly ActivitySummaryCacheLoader _myCacheLoader;
+        private readonly CompositeDisposable _disposable;
+
+        public ActivitySummaryService(IStravaService stravaService, IStravaAthleteService stravaAthleteService)
+        {
+            if (stravaService == null) throw new ArgumentNullException(nameof(stravaService));
+            if (stravaAthleteService == null) throw new ArgumentNullException(nameof(stravaAthleteService));
+
+            _stravaAthleteService = stravaAthleteService;
+
+            _friendsActivitySummaryCache = new SourceCache<ActivitySummary, long>(activitySummary => activitySummary.Id);
+            _myActivitySummaryCache = new SourceCache<ActivitySummary, long>(activitySummary => activitySummary.Id);
+
+            const int pageSize = 30;
+
+            //we have two caches, which we use to load friends (followers and me)...
+            _friendsCacheLoader = new ActivitySummaryCacheLoader(stravaService.GetFriendActivityDataAsync, stravaService.HydrateActivityData, pageSize,
+                _friendsActivitySummaryCache, "All");
+            //..and just me. 
+            _myCacheLoader = new ActivitySummaryCacheLoader(stravaService.GetMyActivityDataAsync, stravaService.HydrateActivityData, pageSize,
+                _myActivitySummaryCache, "My");
+
+            _friendsCacheLoader.LoadMoreItems(pageSize);
+            _myCacheLoader.LoadMoreItems(pageSize);
+
+            _disposable = new CompositeDisposable(_friendsActivitySummaryCache, _myActivitySummaryCache);
+        }
+        
+        public IDisposable Bind(            
+            out DeferringObservableCollection<ActivitySummary> friendsCollection,
+            out DeferringObservableCollection<ActivitySummary> myCollection,
+            out DeferringObservableCollection<ActivitySummary> allCollection
+            )
+        {                        
+            var sortExpressionComparer = SortExpressionComparer<ActivitySummary>.Descending(summary => summary.DateTimeStartLocal);
```

Comment:
@bartlannoeye we have a setting for sorting, is this Descending code on DateTimeStartLocal still correct? Or does @ButchersBoy need to take into account the settings property?